jvm/src/main/scala/org/apache/spark/sql/execution/datasources/v2/BatchScanExec.scala

```diff
@@ -37,24 +43,78 @@ case class BatchScanExec(output: Seq[AttributeReference],
                          runtimeFilters: Seq[Expression]) extends DataSourceV2ScanExecBase {
 
   @transient lazy val batch = scan.toBatch
+
+  // TODO: unify the equal/hashCode implementation for all data source v2 query plans.
+  override def equals(other: Any): Boolean = other match {
```

Comment:
since this is a case class, the equals() func is not required? 
could be cleaned up later

Reply:
it's required, use `batch` and `runtimeFilters` as equal condition, not BatchScanExec.
this code is copied from the Vanilla Spark.